### PR TITLE
Add fix module

### PR DESCRIPTION
**Pull request by m4rc1e**
We have a collection of fix scripts which contain useful snippets of code that should reside in their own module imo. By converting these snippets into functions, we'll get better reuse. There's also a lot of duplication so I'd like to cut down on this.

My end goal is to keep the individual fix scripts (without changing their apis) but I also just want a single `gftools fix-font` script which should do everything.

@alerque are you opposed to this idea?

### CODE REVIEW

**Comment by alerque** [comment]
Opposed? No this sounds like a great idea. Provided the new `gftools fix-font` has a sensible and consistent API itself (with control over the output location) I think that will be a useful thing to have.

I would still like to see the individual fix scripts API fixed such that they all can be used similarly (see e.g. #243 where they all should accept an output arg) ... but that is just an addition not a breakage of the current API.

**Comment by m4rc1e** [comment]
Great. I thought I'd ask for your thoughts since you use these scripts in Fontship.

I've just added the wip `fix-font` script to this PR. Lmk if the input/output is to your liking. Also, would you be keen to review this script/pr?

**Comment by alerque** [comment]
I haven't gotten a chance to take it for a test drive yet, but reviewing the code I link the direction.

I do wonder if there might be a way to pass a list of fixes (either a do list or skip list) to in case some specific fix proves problematic. I don't have any current examples of such a situation, but re-programming to use all the individual fix scrips except the one you need to skip would be a lot harder than adding a single argument to skip something. Just a thought — like I said I haven't gotten to test drive this yet.

File: bin/gftools-fix-font.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+
+log = logging.getLogger(__name__)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("font", help="Path to font")
+    parser.add_argument("-o", "--out", help="Output path for fixed font")
+    parser.add_argument("--hotfix", action="store_true", help="Hotfix fonts.")
```
Comment (alerque) [inline bin/gftools-fix-font.py:21]: I *think* I know what this is doing (applying a certain genre of fixes that *should* be fixed in the sources in addition to the genre of fixes that are rightly post-build cleanup) but the argument itself and help text are kind of misleading. My first impression of `--hotfix` would be that the fixes were being applied inline (i.e. obsoleting the `--out` argument). Is this an established name/concept elsewhere in the tool set or is there perhaps a better name for this?

Comment (m4rc1e) [reply bin/gftools-fix-font.py:21]: > I think I know what this is doing (applying a certain genre of fixes that really ought be fixed in the sources in addition to the genre of fixes that are rightly post-build cleanup and can't be corrected in the sources)

Correct.

> but the argument itself and help text are kind of misleading.

Agreed. I still haven't thought of a decent name for this. First thought was `--force`. Open to suggestions?

> My first impression of --hotfix would be that the fixes were being applied inline (i.e. obsoleting the --out argument). Is this an established name/concept elsewhere in the tool set or is there perhaps a better name for this?

I don't think we have any scripts which inline by default. Personally, I'd still like to be able to run these fixes and not overwrite the existing font, even with this arg applied.

Comment (m4rc1e) [reply bin/gftools-fix-font.py:21]: @alerque I foolishly didn't set this PR to a draft. You'll probably uncover a lot of wtf moments for the time being. I always do a cleanup once I've got a draft down. I'd be grateful for any api suggestions/help though since you're a prominent user.

Comment (alerque) [reply bin/gftools-fix-font.py:21]: Negative on `--force`, that is a very common idiom for for an action that is likely destructive.

How about `--include-source-fixes` or something similarly verbose?

(You know you can convert PR's to Draft mode after the fact right? It's in the "Reviewers" section of the sidebar.)

Comment (m4rc1e) [reply bin/gftools-fix-font.py:21]: `include-source-fixes` is much better. If we can't think of anything better, we'll roll with it. Thanks for the draft convert as well!

Comment (m4rc1e) [reply bin/gftools-fix-font.py]: @alerque, slept on it and I think `--include-source-fixes` will do. Thanks.

**Comment by m4rc1e** [comment]
@alerque I think I'm done.

Please note, I haven't converted all of the fix functions but I believe I have a very good base that covers most of our fix scripts. I've added tests as well because the last scripts didn't have any (I had no clue whether they'd work properly for VFs etc)

I'm still going to look for edge cases and perhaps add better exceptions if needed.

File: bin/gftools-fix-weightclass.py [hunk 2]
```diff
@@ -21,32 +21,17 @@
...
-        elif desired_style.usWeightClass == 200:
-            print(f"{filename}: Updating weightClass to {275}")
-            font['OS/2'].usWeightClass = 275
-            updated = True
-    if updated:
+    fix_weight_class(font)
+    if current_weight_class != font["OS/2"].usWeightClass:
         font.save(font.reader.file.name + ".fix")
     else:
         print("{}: Skipping. Current WeightClass is correct".format(filename))
```
Comment (alerque) [inline bin/gftools-fix-weightclass.py:37]: Not necessarily relevant to this PR, but this is one of the maddeningly inconsistent bits of the old script usages: not only is the output name hard to control, sometimes they produce output and sometimes they don't. You have to run the fix, know what file to expect, check if it exists, rename it if it does, rename the original source file if not.

**Review by alerque — APPROVED** [review]

**Comment by m4rc1e** [comment]
Thanks for the review @alerque. I'll merge this which will then unblock me for #263 and #264.
